src/Reflection/ReflectionConstant.php

```diff
@@ -0,0 +1,308 @@
+     * "FOO").
+     */
+    public function getShortName() : string
+    {
+        if ($this->node instanceof Node\Expr\FuncCall) {
+            $nameParts = explode('\\', $this->getNameFromDefineFunctionCall($this->node));
+
+            return $nameParts[count($nameParts) - 1];
+        }
+
+        /** @psalm-suppress PossiblyNullArrayOffset */
+        return $this->node->consts[$this->positionInNode]->name->name;
+    }
+
+    /**
+     * Get the "full" name of the constant (e.g. for A\B\FOO, this will return
+     * "A\B\FOO").
+     */
+    public function getName() : string
+    {
+        if (! $this->inNamespace()) {
+            return $this->getShortName();
+        }
+
+        if ($this->node instanceof Node\Expr\FuncCall) {
+            return $this->getNameFromDefineFunctionCall($this->node);
+        }
+
+        /**
+         * @psalm-suppress UndefinedPropertyFetch
+         * @psalm-suppress PossiblyNullArrayOffset
+         */
+        return $this->node->consts[$this->positionInNode]->namespacedName->toString();
+    }
+
+    /**
+     * Get the "namespace" name of the constant (e.g. for A\B\FOO, this will
+     * return "A\B").
+     *
+     * @psalm-suppress PossiblyNullPropertyFetch
+     */
+    public function getNamespaceName() : string
+    {
+        if (! $this->inNamespace()) {
+            return '';
+        }
+
+        $namespaceParts = $this->node instanceof Node\Expr\FuncCall
+            ? array_slice(explode('\\', $this->getNameFromDefineFunctionCall($this->node)), 0, -1)
+            : $this->declaringNamespace->name->parts;
+
+        return implode('\\', $namespaceParts);
+    }
+
+    /**
+     * Decide if this constant is part of a namespace. Returns false if the constant
+     * is in the global namespace or does not have a specified namespace.
+     */
+    public function inNamespace() : bool
+    {
+        if ($this->node instanceof Node\Expr\FuncCall) {
+            return substr_count($this->getNameFromDefineFunctionCall($this->node), '\\') !== 0;
+        }
+
+        return $this->declaringNamespace !== null
+            && $this->declaringNamespace->name !== null;
+    }
+
+    public function getExtensionName() : ?string
+    {
+        return $this->locatedSource->getExtensionName();
+    }
+
+    /**
+     * Is this an internal constant?
+     */
+    public function isInternal() : bool
+    {
+        return $this->locatedSource->isInternal();
+    }
+
+    /**
+     * Is this a user-defined function (will always return the opposite of
+     * whatever isInternal returns).
+     */
+    public function isUserDefined() : bool
+    {
+        return ! $this->isInternal();
+    }
+
+    /**
+     * Returns constant value
+     *
+     * @return mixed
+     */
+    public function getValue()
+    {
+        if ($this->valueWasCached !== false) {
+            return $this->value;
+        }
+
+        /** @psalm-suppress PossiblyNullArrayOffset */
```

Comment:
This is a leap of faith being done: would it be better for the utility that inspects constant declarations to already return the parameters?

Reply:
It cannot be `null` because if the node is `Node\Stmt\Const_` then the `ReflectionConstant` is created by `createFromConstKeyword` where `positionInNode` is not nullable.